### PR TITLE
Centered editor layout

### LINKED ISSUE #50936: Grid: Implement "Centered Editor Layout"
We should restore the functionality we had. Ideally the code we used to have could be encapsulated into a widget that has these features:
* sashes to the left and right with a container in the center
* sashes move on both sides if one side is dragged
* some way to persist and restore view state

Having this widget, we can throw it into an editor group view based on the condition that centered editor layout is enabled or not.

### PR DESCRIPTION
**Pull request by isidorn**
Fixes #50936

Brings back centered editor layout support to VSCode.
This is work in progress, creating PR now to potentially get early feedback.

Still needs to be done:
- [x] double check layout of the grid - seems to overflow to the right
- [x] add separator style (due to overflow right separtor is hidden)
- [x] background color of left and right empty areas should have the color of the editor

### CODE REVIEW

**Comment by joaomoreno** [comment]
@isidorn I would merged `master` into this, since there was an ugly issue which I just fixed this morning.

**Comment by isidorn** [comment]
I have merged master.
I am now also centering the layout by pushing empty views, not reparenting html elements - so it is more elegent.

Still needs some polish from my first comment, once I address that I plan to merge this in if nobody objects

File: src/vs/workbench/browser/parts/editor/editorPart.ts
Comment (joaomoreno) [inline src/vs/workbench/browser/parts/editor/editorPart.ts]: Isn't there always a `centeredViewLayout`? This will always be true, it seems.

File: src/vs/workbench/browser/parts/editor/editorPart.ts
Comment (joaomoreno) [inline src/vs/workbench/browser/parts/editor/editorPart.ts]: This looks super weird. There must always be a `this.dimension`. It's impossible for this view to be asked for layout before `doLayout` has run.

Also, if you want to remove the weirdness, don't simply use `IView` from splitview. Use `IView` from gridview. This one talks about `width` and `height`. Simultaneously, `CenteredViewLayout.layout` would take in both `width` and `height`.

File: src/vs/base/browser/ui/centered/centeredViewLayout.ts
Comment (joaomoreno) [inline src/vs/base/browser/ui/centered/centeredViewLayout.ts]: You probably also need `if (this.active === active) { return; }` before this.

File: src/vs/base/browser/ui/centered/centeredViewLayout.ts
Comment (joaomoreno) [inline src/vs/base/browser/ui/centered/centeredViewLayout.ts]: You need to add two different views... How can that DOM element be put twice in the DOM?

File: src/vs/base/browser/ui/centered/centeredViewLayout.ts
Comment (joaomoreno) [inline src/vs/base/browser/ui/centered/centeredViewLayout.ts]: `$('.')` doesn't mean anything. If you wan a div, use `$('div')`. Though you probably should use something descriptive like `$('.centered-layout-margin')`.

File: src/vs/base/browser/ui/centered/centeredViewLayout.ts
Comment (joaomoreno) [inline src/vs/base/browser/ui/centered/centeredViewLayout.ts]: Why not disposing the splitview altogether and just put the provided view directly as a child of the container?

File: src/vs/workbench/browser/parts/editor/editorPart.ts
Comment (joaomoreno) [inline src/vs/workbench/browser/parts/editor/editorPart.ts]: All you need is `this.centeredViewLayout.dispose()`.

**Review by joaomoreno — CHANGES_REQUESTED** [review]

**Comment by isidorn** [comment]
@joaomoreno thanks for the review, I have addressed your feedback. There are still some nit-bits to take care of.

What I find a bit ugly is that for some methods the `editorPart` goes directly to the `gridWidget` and for some it goes to the `centeredLayout` instead which forwards to the `gridWidget`.

File: src/vs/base/browser/ui/centered/centeredViewLayout.ts
Comment (joaomoreno) [inline src/vs/base/browser/ui/centered/centeredViewLayout.ts]: You should expose `this.view.onDidChange` here.

File: src/vs/base/browser/ui/centered/centeredViewLayout.ts
Comment (joaomoreno) [inline src/vs/base/browser/ui/centered/centeredViewLayout.ts]: There is `splitview.distributeViewSizes()`.

File: src/vs/base/browser/ui/centered/centeredViewLayout.ts
Comment (joaomoreno) [inline src/vs/base/browser/ui/centered/centeredViewLayout.ts]: You should still make a `if (active === !!this.splitview) { return; }`, so you don't get any NPEs.

**Review by joaomoreno — CHANGES_REQUESTED** [review]

**Comment by isidorn** [comment]
@joaomoreno thanks, I have addressed that feedback in the latest commit.
There are still some issues with the watermark that I want to address before merging in

**Pull request merged**
